<ui version="4.0" >
 <class>OpenVPNProp</class>
 <widget class="QWidget" name="OpenVPNProp" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>686</width>
    <height>369</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>OpenVPN</string>
  </property>
  <layout class="QGridLayout" name="_2" >
   <item row="0" column="0" >
    <widget class="QTabWidget" name="kTabWidget1" >
     <property name="currentIndex" >
      <number>0</number>
     </property>
     <widget class="QWidget" name="TabPage" >
      <attribute name="title" >
       <string>Required Information</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5" >
       <item>
        <layout class="QGridLayout" name="gridLayout_5" >
         <item row="0" column="0" >
          <widget class="QLabel" name="textLabel3" >
           <property name="text" >
            <string>&amp;Gateway</string>
           </property>
           <property name="wordWrap" >
            <bool>false</bool>
           </property>
           <property name="buddy" >
            <cstring>gateway</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1" >
          <widget class="KLineEdit" name="gateway" />
         </item>
         <item row="1" column="0" >
          <widget class="QLabel" name="textLabel3_3" >
           <property name="text" >
            <string>&amp;Port</string>
           </property>
           <property name="wordWrap" >
            <bool>false</bool>
           </property>
           <property name="buddy" >
            <cstring>port</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1" >
          <layout class="QHBoxLayout" name="horizontalLayout" >
           <item>
            <widget class="KLineEdit" name="port" >
             <property name="enabled" >
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="chkDefaultPort" >
             <property name="text" >
              <string>Use &amp;Default</string>
             </property>
             <property name="checked" >
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0" >
          <widget class="QLabel" name="labelConnectionType" >
           <property name="text" >
            <string>Connection &amp;Type</string>
           </property>
           <property name="wordWrap" >
            <bool>false</bool>
           </property>
           <property name="buddy" >
            <cstring>cmbConnectionType</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1" >
          <widget class="KComboBox" name="cmbConnectionType" >
           <item>
            <property name="text" >
             <string>X.509 Certificates</string>
            </property>
           </item>
           <item>
            <property name="text" >
             <string>Pre-shared Key</string>
            </property>
           </item>
           <item>
            <property name="text" >
             <string>Password</string>
            </property>
           </item>
           <item>
            <property name="text" >
             <string>X.509 With Password</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QStackedWidget" name="stackedWidget" >
         <property name="currentIndex" >
          <number>1</number>
         </property>
         <widget class="QWidget" name="page" >
          <layout class="QVBoxLayout" name="verticalLayout" >
           <item>
            <layout class="QGridLayout" name="gridLayout_4" >
             <item row="0" column="0" >
              <widget class="QLabel" name="textLabel2" >
               <property name="text" >
                <string>C&amp;A file</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>x509CaFile</cstring>
               </property>
              </widget>
             </item>
             <item row="0" column="1" >
              <widget class="KUrlRequester" name="x509CaFile" />
             </item>
             <item row="1" column="0" >
              <widget class="QLabel" name="textLabel3_2" >
               <property name="text" >
                <string>&amp;Certificate</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>x509Cert</cstring>
               </property>
              </widget>
             </item>
             <item row="1" column="1" >
              <widget class="KUrlRequester" name="x509Cert" />
             </item>
             <item row="2" column="0" >
              <widget class="QLabel" name="textLabel4" >
               <property name="text" >
                <string>&amp;Key</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>x509Key</cstring>
               </property>
              </widget>
             </item>
             <item row="2" column="1" >
              <widget class="KUrlRequester" name="x509Key" />
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer" >
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>87</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_2" >
          <layout class="QVBoxLayout" name="verticalLayout_2" >
           <item>
            <layout class="QGridLayout" name="gridLayout" >
             <item row="0" column="0" >
              <widget class="QLabel" name="textLabel5" >
               <property name="text" >
                <string>Shared Key</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>pskSharedKey</cstring>
               </property>
              </widget>
             </item>
             <item row="0" column="1" >
              <widget class="KUrlRequester" name="pskSharedKey" />
             </item>
             <item row="1" column="0" >
              <widget class="QLabel" name="textLabel6" >
               <property name="text" >
                <string>Local IP</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>pskLocalIp</cstring>
               </property>
              </widget>
             </item>
             <item row="1" column="1" >
              <widget class="KLineEdit" name="pskLocalIp" />
             </item>
             <item row="2" column="0" >
              <widget class="QLabel" name="textLabel7" >
               <property name="text" >
                <string>Remote IP</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>pskRemoteIp</cstring>
               </property>
              </widget>
             </item>
             <item row="2" column="1" >
              <widget class="KLineEdit" name="pskRemoteIp" >
               <property name="text" >
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_2" >
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>85</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_3" >
          <layout class="QVBoxLayout" name="verticalLayout_3" >
           <item>
            <layout class="QGridLayout" name="gridLayout_2" >
             <item row="0" column="0" >
              <widget class="QLabel" name="textLabel8" >
               <property name="text" >
                <string>CA file</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>passCaFile</cstring>
               </property>
              </widget>
             </item>
             <item row="0" column="1" >
              <widget class="KUrlRequester" name="passCaFile" />
             </item>
             <item row="1" column="0" >
              <widget class="QLabel" name="textLabel9" >
               <property name="text" >
                <string>Username</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>passUserName</cstring>
               </property>
              </widget>
             </item>
             <item row="1" column="1" >
              <widget class="KLineEdit" name="passUserName" />
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_3" >
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>120</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_4" >
          <layout class="QVBoxLayout" name="verticalLayout_4" >
           <item>
            <layout class="QGridLayout" name="gridLayout_3" >
             <item row="0" column="0" >
              <widget class="QLabel" name="textLabel2_2" >
               <property name="text" >
                <string>CA file</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>x509PassCaFile</cstring>
               </property>
              </widget>
             </item>
             <item row="0" column="1" >
              <widget class="KUrlRequester" name="x509PassCaFile" />
             </item>
             <item row="1" column="0" >
              <widget class="QLabel" name="textLabel3_2_2" >
               <property name="text" >
                <string>Certificate</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>x509PassCert</cstring>
               </property>
              </widget>
             </item>
             <item row="1" column="1" >
              <widget class="KUrlRequester" name="x509PassCert" />
             </item>
             <item row="2" column="0" >
              <widget class="QLabel" name="textLabel4_2" >
               <property name="text" >
                <string>Key</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>x509PassKey</cstring>
               </property>
              </widget>
             </item>
             <item row="2" column="1" >
              <widget class="KUrlRequester" name="x509PassKey" />
             </item>
             <item row="3" column="0" >
              <widget class="QLabel" name="textLabel9_2" >
               <property name="text" >
                <string>Username</string>
               </property>
               <property name="wordWrap" >
                <bool>false</bool>
               </property>
               <property name="buddy" >
                <cstring>x509PassUsername</cstring>
               </property>
              </widget>
             </item>
             <item row="3" column="1" >
              <widget class="KLineEdit" name="x509PassUsername" />
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_4" >
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>52</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="TabPage" >
      <attribute name="title" >
       <string>Optional Information</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_6" >
       <item row="0" column="0" colspan="3" >
        <widget class="QCheckBox" name="chkUseLZO" >
         <property name="text" >
          <string>Use LZO compression</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="3" >
        <widget class="QCheckBox" name="chkUseTAP" >
         <property name="text" >
          <string>Use TAP device</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2" >
        <widget class="QCheckBox" name="chkUseTCP" >
         <property name="text" >
          <string>Use TCP connection</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" >
        <widget class="QCheckBox" name="chkUseCipher" >
         <property name="text" >
          <string>Use cipher</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1" colspan="2" >
        <widget class="KComboBox" name="cboCipher" >
         <property name="enabled" >
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2" >
        <widget class="QCheckBox" name="chkUseTLS" >
         <property name="text" >
          <string>Use TLS auth</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2" >
        <widget class="KUrlRequester" name="editTLSAuth" >
         <property name="enabled" >
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="0" >
        <widget class="QLabel" name="textLabel1" >
         <property name="text" >
          <string>Direction:</string>
         </property>
         <property name="wordWrap" >
          <bool>false</bool>
         </property>
         <property name="buddy" >
          <cstring>cboDirection</cstring>
         </property>
        </widget>
       </item>
       <item row="5" column="1" colspan="2" >
        <widget class="KComboBox" name="cboDirection" >
         <property name="enabled" >
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="6" column="0" colspan="2" >
        <widget class="QCheckBox" name="chkIPAdresses" >
         <property name="text" >
          <string>Only use VPN connection for these addresses (eg: 172.16.0.0/16 10.11.12.0/24)</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="3" >
        <widget class="KLineEdit" name="routes" >
         <property name="enabled" >
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="8" column="0" colspan="2" >
        <spacer name="spacer10" >
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType" >
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>20</width>
           <height>30</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>chkUseLZO</tabstop>
  <tabstop>chkUseTAP</tabstop>
  <tabstop>chkUseTCP</tabstop>
  <tabstop>chkUseCipher</tabstop>
  <tabstop>cboCipher</tabstop>
  <tabstop>chkUseTLS</tabstop>
  <tabstop>editTLSAuth</tabstop>
  <tabstop>cboDirection</tabstop>
  <tabstop>routes</tabstop>
  <tabstop>gateway</tabstop>
  <tabstop>cmbConnectionType</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>chkDefaultPort</sender>
   <signal>toggled(bool)</signal>
   <receiver>port</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>872</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel" >
     <x>129</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbConnectionType</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>528</x>
     <y>130</y>
    </hint>
    <hint type="destinationlabel" >
     <x>584</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
